tests/foreman/cli/test_host.py

```diff
@@ -2076,6 +2078,188 @@ def test_negative_multi_cv_registration(
+
+
+@pytest.mark.rhel_ver_match('[^7]')
+def test_positive_multi_cv_host_repo_availability(
+    session_multicv_sat,
+    rhel_contenthost,
+):
+    """Multi-environment hosts should have access to repositories from all of their content view environments.
+
+    :id: 6a15d591-be84-4b5b-8deb-8ea6eb32fee6
+
+    :steps:
+        1. Create two synced custom repositories
+        2. Add each repository to its own content view. Publish and promote the content views.
+        3. Register a host with global registration to a single content view environment
+        4. Expose the allow_multiple_content_views setting
+        5. Set the allow_multiple_content_views setting to true
+        6. Assign the host to multiple content view environments
+        7. Confirm repos listed in subscription-manager repos match the repos in the content view environments
+
+    :expectedresults: Host sees repositories from both content view environments, not just one.
+
+    :CaseImportance: Critical
+
+    :parametrized: yes
+    """
+
+    # Create an org
+    multicv_org = session_multicv_sat.api.Organization().create()
+
+    # Create activation key
+    multicv_ak = session_multicv_sat.api.ActivationKey(
+        content_view=multicv_org.default_content_view.id,
+        environment=multicv_org.library.id,
+        organization=multicv_org,
+    ).create()
+
+    # Create a lifecycle environment
+    multicv_lce = session_multicv_sat.api.LifecycleEnvironment(organization=multicv_org).create()
+
+    # Create a content view
+    cv1 = session_multicv_sat.api.ContentView(organization=multicv_org).create()
+    cv1.publish()
+    cv1 = cv1.read()
+
+    # Create two synced custom repositories
+    repo_instances = []
+    for repo in ['RepoA', 'RepoB']:
+        product_name = f'{repo}_Product_{str(uuid.uuid4())}'
+        repo_id = session_multicv_sat.api_factory.create_sync_custom_repo(
+            org_id=multicv_org.id,
+            product_name=product_name,
+            repo_name=repo,
+            repo_url=settings.repos.fake_repo_zoo3,
+        )
+        repo_instances.append(session_multicv_sat.api.Repository(id=repo_id).read())
+
+    repo_b = repo_instances.pop()
+    repo_a = repo_instances.pop()
+
+    # Add one repository each to two content views. Publish and promote the content views.
+    cv1.repository = [repo_a]
+    cv1 = cv1.update(['repository'])
+    cv1.publish()
+    # for some reason I have to call read() or it doesn't pick up the new version
+    module_published_cvv = cv1.read().version[0]
+    module_published_cvv.promote(data={'environment_ids': multicv_lce.id})
+
+    lce2 = session_multicv_sat.api.LifecycleEnvironment(organization=multicv_org).create()
+
+    cv2 = session_multicv_sat.api.ContentView(organization=multicv_org).create()
+    cv2.publish()
+    cv2.repository = [repo_b]
+    cv2 = cv2.update(['repository'])
+    cv2.publish()
+    cv2_content_view_version = cv2.read().version[0]
```

Comment:
This is a bit tricky. Now we have two versions, one is empty. Is the `version[0]` the latest one?

For context - there was an issue (unsorted CV versions returned by API) which required us to sort on our side like [here](https://github.com/SatelliteQE/robottelo/blob/1e952b13f3ed098bf8063c52703e8a4065d79619/pytest_fixtures/component/repository.py#L52-L55)

Reply:
Neither version should be empty. One version should contain RepoA and the other should contain RepoB. I am trying to show that the host has access to both repos, by being assigned to both content views.

Reply:
I meant CV version - after first `publish()` (L2230) you get "Version 1.0" in that CV, which is empty. After update and second `publish()` (L2233) you get "Version 2.0" in that CV, which has the repo.
`cv2.read().version` returns both in a list and `[0]` wasn't always the latest

Reply:
Anyway if you go with `ContentView(organization=multicv_org, repository=[repo_b]).create()` there will be only "Version 1.0" and we can mark this as resolved :)

Reply:
Ah. Looks like Katello sorts them now:
```rb
# app/views/katello/api/v2/content_views/base.json.rabl
child :sorted_versions => :versions do

# app/models/katello/content_view.rb
def sorted_versions
  versions.order('created_at DESC')
end
```